### PR TITLE
Add code example to documentation around `before_delete_page` hook.

### PR DESCRIPTION
**Pull request by janeliutw**
issue #6499. Let me know if anything needs to be updated!

Thanks for contributing to Wagtail! 🎉

Before submitting, please review the contributor guidelines <https://docs.wagtail.io/en/latest/contributing/index.html> and check the following:

* Do the tests still pass? (https://docs.wagtail.io/en/latest/contributing/developing.html#testing)
* Does the code comply with the style guide? (Run `make lint` from the Wagtail root)
* For Python changes: Have you added tests to cover the new/fixed behaviour?
* For front-end changes: Did you test on all of Wagtail’s supported browsers? **Please list the exact versions you tested**.
* For new features: Has the documentation been updated accordingly?

### CODE REVIEW

File: docs/reference/hooks.rst [hunk 1]
```diff
@@ -533,6 +533,16 @@ Hooks for customising the way users are directed through the process of creating
...
+ .. code-block:: python
+
+    from wagtail.core import hooks
+
+    from .models import AwesomePage
+    from .admin_views import edit_awesome_page
+
+    @hooks.register('before_delete_page')
+    def before_delete_page(request, page):
+        return delete_awesome_page(request, page)
```
Comment (tomdyson) [inline docs/reference/hooks.rst:545]: I know this is also in the `before_create_page` example, but I'm not sure why we import `edit_awesome_page` then call `delete_awesome_page`.

**Review by tomdyson — COMMENTED** [review]
Thanks @janeliutw!

File: docs/reference/hooks.rst (hunk 1 above)
Comment (janeliutw) [reply docs/reference/hooks.rst:545]: Yeah I was actually wondering about that as well, but I am not super familiar with the project so I figured I just kept it. Happy to remove it in both of the examples though!

**Comment by lb-** [comment]
@janeliutw  thanks so much for this PR, I know it has been a long while but we have now merged this in with a bit of refinement via 92381698d0215d9f26c71a52c61ddeb429fb96e7

Really appreciate your time taken on this, thank you for contributing to Wagtail.

**Pull request closed without merging**